sds_data_manager/lambda_code/SDSCode/path_helper.py

```diff
@@ -58,14 +44,18 @@ def __init__(self, filename):
         Parameters
         ----------
         filename : str
-            Filename
+            Science data filename
         """
         self.filename = filename
-        self.filename_convention = (
+        self.split_filename = self.filename.split("_")
+        filename_convention = (
             "<mission>_<instrument>_<datalevel>_<descriptor>_"
             "<startdate>_<enddate>_<version>.<extension>"
         )
-        self.split_filename = filename.split("_")
+
+        if len(self.split_filename) != 7:
+            raise ValueError(f"Invalid filename. Expected - {filename_convention}")
```

Comment:
If you were to take my other try/except suggestion (you don't need to, it is only an idea/suggestion), then this would be:
```suggestion
        if len(self.split_filename) != 7:
            raise InvalidScienceFile(f"Invalid filename. Expected - {filename_convention}")
```